rpcs3/rpcs3qt/cheat_manager.cpp

```diff
@@ -0,0 +1,660 @@
+		offsets_found.clear();
+		do_the_search();
+	});
+
+	connect(btn_filter_results, &QPushButton::clicked, [=](int action) { do_the_search(); });
+
+	connect(lst_search, &QListWidget::customContextMenuRequested, [=](const QPoint& loc) {
+		QPoint globalPos      = lst_search->mapToGlobal(loc);
+		QMenu* menu           = new QMenu();
+		QListWidgetItem* item = lst_search->item(lst_search->currentRow());
+		if (!item)
+			return;
+
+		QAction* add_to_cheat_list = new QAction(tr("Add to cheat list"), menu);
+
+		u32 offset       = offsets_found[lst_search->currentRow()];
+		u8 type          = cbx_cheat_search_type->currentIndex();
+		std::string name = Emu.GetTitle();
+
+		connect(add_to_cheat_list, &QAction::triggered, [=]() {
+			if (g_cheat.exist(name, offset))
+			{
+				if (QMessageBox::question(this, tr("Cheat already exist"), tr("Do you want to overwrite the existing cheat?"), QMessageBox::Ok | QMessageBox::Cancel) != QMessageBox::Ok)
+					return;
+			}
+
+			g_cheat.add(name, "", type, offset);
+			update_cheat_list();
+		});
+
+		menu->addAction(add_to_cheat_list);
+		menu->exec(globalPos);
+	});
+
+	update_cheat_list();
+}
+
+void cheat_manager_dialog::do_the_search()
+{
+	bool res_conv;
+	QString qstr_to_search = edt_cheat_search_value->text();
+
+	// TODO: better way to do this?
+	switch (cbx_cheat_search_type->currentIndex())
+	{
+	case unsigned_8_cheat:
+	{
+		u16 to_search = qstr_to_search.toUShort(&res_conv);
+		if (!res_conv || to_search > 0xFF)
+		{
+			res_conv = false;
+			break;
+		}
+		offsets_found = g_cheat.search((u8)to_search, offsets_found);
+		break;
+	}
+	case unsigned_16_cheat:
+	{
+		u16 to_search = qstr_to_search.toUShort(&res_conv);
+		if (res_conv)
+			offsets_found = g_cheat.search(to_search, offsets_found);
+		break;
+	}
+	case unsigned_32_cheat:
+	{
+		u32 to_search = qstr_to_search.toULong(&res_conv);
+		if (res_conv)
+			offsets_found = g_cheat.search(to_search, offsets_found);
+		break;
+	}
+	case unsigned_64_cheat:
+	{
+		u64 to_search = qstr_to_search.toULongLong(&res_conv);
+		if (res_conv)
+			offsets_found = g_cheat.search(to_search, offsets_found);
+		break;
+	}
+	case signed_8_cheat:
+	{
+		s16 to_search = qstr_to_search.toShort(&res_conv);
+		if (!res_conv || to_search < -128 || to_search > 127)
+		{
+			res_conv = false;
+			break;
+		}
+		offsets_found = g_cheat.search((s8)to_search, offsets_found);
+		break;
+	}
+	case signed_16_cheat:
+	{
+		s16 to_search = qstr_to_search.toShort(&res_conv);
+		if (res_conv)
+			offsets_found = g_cheat.search(to_search, offsets_found);
+		break;
+	}
+	case signed_32_cheat:
+	{
+		s32 to_search = qstr_to_search.toLong(&res_conv);
+		if (res_conv)
+			offsets_found = g_cheat.search(to_search, offsets_found);
+		break;
+	}
+	case signed_64_cheat:
+	{
+		s64 to_search = qstr_to_search.toLongLong(&res_conv);
+		if (res_conv)
+			offsets_found = g_cheat.search(to_search, offsets_found);
+		break;
+	}
+	}
+
+	if (!res_conv)
```

Comment:
```suggestion
	if (res_conv)
	{
		offsets_found = g_cheat.search(to_search, offsets_found);
	}
	else
```

Reply:
I can't do that because search is a template and I need to specify or let the compiler deduce the template type.